src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedPhysicalPlanPrinterImpl2.java

```diff
@@ -0,0 +1,293 @@
+		final LogicalOperator lop = op.getLogicalOperator();
+		printLogicalOperatorBase( "  - lop ", op.getLogicalOperator(), out, indentLevelStringForOpDetail + singleBase );
+		out.append( System.lineSeparator() );
+		printOperatorInfoFmAndPatternWithOpt( lop, out, indentLevelStringForOpDetail );
+	}
+
+	protected void printOperatorInfoForOpLocalToGlobal ( final PhysicalOpLocalToGlobal op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "l2g (" + op.getID() + ") " );
+		out.append( System.lineSeparator() );
+		final LogicalOperator lop = op.getLogicalOperator();
+		printLogicalOperatorBase( "  - lop ", lop, out, indentLevelStringForOpDetail + singleBase );
+		out.append( "  vocab.mapping (" + ((LogicalOpGlobalToLocal) lop).getVocabularyMapping().hashCode() + ")" );
+		out.append( System.lineSeparator() );
+	}
+	
+	protected void printOperatorInfoForMultiwayUnion ( final PhysicalOpMultiwayUnion op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "multiwayUnion (" + op.getID() + ") " );
+		out.append( System.lineSeparator() );
+		printLogicalOperatorBase( "  - lop ", op.getLogicalOperator(), out, indentLevelStringForOpDetail + singleBase );
+		out.append( System.lineSeparator() );
+	}
+
+	protected void printOperatorInfoForOpNaiveNestedLoopsJoin ( final PhysicalOpNaiveNestedLoopsJoin op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "naiveNestedLoop (" + op.getID() + ") " );
+		out.append( System.lineSeparator() );
+		printLogicalOperatorBase( "  - lop ", op.getLogicalOperator(), out, indentLevelStringForOpDetail + singleBase );
+		out.append( System.lineSeparator() );
+	}
+
+	protected void printOperatorInfoForOpParallelMultiLeftJoin ( final PhysicalOpParallelMultiLeftJoin op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "parallelMultiLeftJoin (" + op.getID() + ") " );
+		out.append( System.lineSeparator() );
+	}
+	
+	protected void printOperatorInfoForRequest( final PhysicalOpRequest op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		final LogicalOpRequest lop = op.getLogicalOperator();
+		final DataRetrievalRequest req = lop.getRequest();
+		out.append( indentLevelString + "req (" + op.getID() + ")" );
+		out.append( System.lineSeparator() );
+		printFederationMember( lop.getFederationMember(), indentLevelStringForOpDetail, out );
+		out.append( indentLevelStringForOpDetail + "  - pattern (" + req.hashCode() +  ") (" + req.toString() + ")" );
+		out.append( System.lineSeparator() );
+		printLogicalOperatorBase( "  - lop ", lop, out, indentLevelStringForOpDetail );
+		out.append( System.lineSeparator() );
+	}
```

Comment:
In this one, the order of `- fm`+`- pattern` vs `- lop` is different than in all the other cases. Keep it consistent.